### ISSUE TITLE
Re-implement minor features that were lost when removing the Search section?

### ISSUE THREAD

**Issue by christophfriedrich** [Open-EO/openeo-hub#47]
These minor features were lost when removing the Search section (#42):

1. Filtering for specific openEO API versions, down to the minor version number (i.e. for 0.4.2 instead of 0.4.x)
2. Searching within collection/process metadata
   a. Searching within collection descriptions (only id and title searchable in filter feature)
   b. Searching within process descriptions (only id and summary searchable in filter feature)
   c. Regex support for searching within the fields (id, title/summary, description) of collections/processes
   d. Sophisticated full-text search for all these fields (word-stemming etc.)
3. "Exclude processes that are deprecated" switch
4. Searching within processes' parameter names and descriptions

IMO:
1 was changed this way on purpose.
The things under 2 would be nice-to-have, but are not really necessary.
3 should be added again.
4 is obsolete (was never _really_ needed anyway).

Any other opinions?

**Comment by m-mohr** [comment]
Agreed, re-implementing any of the options is not a priority at the moment and thus assigning it for 1.0. First thing to add again would be 3 and then 2. For 3 we would need to define the single source of truth (similar issues as for the titles/summaries).
